File: ExpenseManager/EM.API/Models/Report.cs
using System;
using System.Collections.Generic;

namespace EM.API.Models
{
    public class Report
    {
        public Guid CategoryID { get; set; }
        public string CategoryName { get; set; }
        public decimal Percentage { get; set; }
        public decimal Amount { get; set; }
        public List<Transaction> Transactions { get; set; }
    }
}

File: ExpenseManager/EMWEB/src/app/models/report.ts
import { Transaction } from "./Transaction";

export class Report {
    CategoryID: string;
    CategoryName: string;
    Percentage: number;
    Amount: number;
    Transactions: Transaction[];

    constructor(object?:any) {
        if(object) {
            Object.assign(this, object)
        }
    }
}  